### PR TITLE
refactor: move segment manifest cache to fetch module

### PR DESCRIPTION
**Pull request by jeqo**
Context: There were some previous efforts on defining generic cache config. This follows up on that by making the SegmentManifest fetching reuse this definition. It also aligns with the SegmentIndexes fetching.

Moving the segment-manifest fetching to the fetch module helps to align with other caches and reuse configuration.

[Following work](https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/591) should help to align cache config for chunk caching.

### CODE REVIEW

File: core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java:1]: Expanding to make it compatible with RSMConfig defaults

File: core/src/test/java/io/aiven/kafka/tieredstorage/config/CacheConfigTest.java
Comment (jeqo) [inline core/src/test/java/io/aiven/kafka/tieredstorage/config/CacheConfigTest.java:1]: Cover all defaults config tests + new set default option

File: core/src/test/java/io/aiven/kafka/tieredstorage/config/RemoteStorageManagerConfigTest.java
Comment (jeqo) [inline core/src/test/java/io/aiven/kafka/tieredstorage/config/RemoteStorageManagerConfigTest.java:1]: Handled by CacheConfigTest already

File: core/src/main/java/io/aiven/kafka/tieredstorage/fetch/manifest/MemorySegmentManifestCache.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023 Aiven Oy
```
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/fetch/manifest/MemorySegmentManifestCache.java:2]: not new, but moving+renaming SegmentManifestProvider

**Review by biggusdonzus — APPROVED** [review]
:+1:

**Pull request merged**